src/JSON.ts

```diff
@@ -34,20 +35,28 @@ export enum ValueKind {
  * The different types of tokens that can be parsed from a JSON string.
```

Comment:
Functionality changes in this file are mainly:
1) add strict and "enclosed" behavior for finding tokens - the current behavior is wonky ("extended" - if the position is right after a token, it still counts as being on the token) which was intended for completion behavior.  I needed behavior to help determine where parameters can be added based on what object the position is "inside"
2) better functionality for handing comments